### PR TITLE
Django 1.8 support

### PR DESCRIPTION
**Pull request by kaedroho**
Addresses #284 

This brings Taggit up to speed with Django 1.8 beta 2. There may be more work to do before the final release but this should get us most of the way there.

### CODE REVIEW

**Comment by kaedroho** [comment]
All tests are now passing. Travis reports an error with Django 1.8/Python 2.7. This appears to be an issue with installing Django and I've sucessfully run the tests locally on Python 2.7/Django 1.8.

**Comment by collinanderson** [comment]
Interesting. The 1.8 issue might be this https://github.com/django/django/commit/bd059e3f8c6311dcaf8afe5e29ef373f7f84cf26#commitcomment-10220493

Maybe switching from `master.zip` to `master.tar.gz` might fix it?

**Comment by collinanderson** [comment]
We could also change the tag_model property to use `get_field()` on 1.8 so it doesn't raise a deprecation warning about `get_field_by_name`.

**Comment by collinanderson** [comment]
I started switching the `get_field_by_name`'s but there's too many :). Probably out of scope for this PR.

**Comment by collinanderson** [comment]
In any case, I tested this, and it at least works on _my_ site :)

File: taggit/managers.py [hunk 1]
```diff
@@ -309,13 +339,18 @@ def __lt__(self, other):
         return False
 
     def post_through_setup(self, cls):
-        self.related = RelatedObject(cls, self.model, self)
+        if RelatedObject is not None:  # Django < 1.8
```
Comment (dtiesling) [inline taggit/managers.py]: I'm able to import `RelatedObject` in django 1.8b2 so this logic is no longer working. Could you use django.VERSION for this instead?

Comment (kaedroho) [inline taggit/managers.py]: related.py is gone in the latest code: https://github.com/django/django/tree/stable/1.8.x/django/db/models

Maybe it's a ``.pyc`` file hanging around, could you try reinstalling Django?

Comment (collinanderson) [inline taggit/managers.py]: Interesting. If @dtiesling is running into that problem, I bet others will too. I wonder if we should use the VERSION method just to be safe.

File: .travis.yml [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
+   - env: TOXENV=py33-1.7.x
+   - env: TOXENV=py33-1.8.x
+   - env: TOXENV=py34-1.5.x
+   - env: TOXENV=py34-1.6.x
+   - env: TOXENV=py34-1.7.x
+   - env: TOXENV=py34-1.8.x
+  allow_failures:
+   - env: TOXENV=py27-1.8.x
+   - env: TOXENV=py33-1.8.x
+   - env: TOXENV=py34-1.8.x
```
Comment (collinanderson) [inline .travis.yml]: Do you want to remove or revert the allow_failures commit?

Comment (kaedroho) [inline .travis.yml]: Yes, remove it. Thanks

**Comment by collinanderson** [comment]
Also, if you didn't notice, it looks like you need to bring the .tar.gz change back.

**Comment by kaedroho** [comment]
Oops, thanks for spotting!

File: taggit/managers.py (hunk 1 above)
Comment (dtiesling) [inline taggit/managers.py]: thanks @kaedroho, you were right it was leftover files from not doing a clean install of django. I agree with @collinanderson I'm sure I won't be the only person to not think to clear out the old django before upgrading.

Comment (kaedroho) [inline taggit/managers.py]: Fixed in 16f9f4a

**Comment by collinanderson** [comment]
This looks good to me, at least :)

**Comment by frewsxcv** [comment]
I'll merge this as soon as 1.8 final is out and i'll push a new taggit release.

**Comment by chhantyal** [comment]
As 1.8 is released (https://www.djangoproject.com/weblog/2015/apr/01/release-18-final/), can we please accept this PR and update PyPi?

**Comment by frewsxcv** [comment]
Has anyone tested this with the final release of 1.8?  The new version came out when I was on a flight. I'll look at merging this tomorrow.

**Comment by frewsxcv** [comment]
Reopened to rerun the travis tests

**Comment by troygrosfield** [comment]
:+1:

**Comment by frewsxcv** [comment]
Great work @kaedroho! Thanks! Here is cake

**Comment by frewsxcv** [comment]
0.13.0 has been published

https://pypi.python.org/pypi/django-taggit/0.13.0

let me know if any issues arise

**Pull request merged**